### PR TITLE
Adding jax.numpy.histogram

### PR DESCRIPTION
**Pull request by marwanWaly**
I wrote everything as it should be according to docs and other functions in the same module but I have some problems:

1) I cannot run other unit tests in the same module, although I was able to run them before my last pull.
when I try I face this error:
`ImportError while loading conftest '/opt/project/ivy_tests/test_ivy/conftest.py'.
../../../conftest.py:7: in <module>
    import ivy_tests.test_ivy.helpers.test_parameter_flags as pf
../../../helpers/__init__.py:2: in <module>
    from . import hypothesis_helpers
../../../helpers/hypothesis_helpers/__init__.py:1: in <module>
    from . import general_helpers
../../../helpers/hypothesis_helpers/general_helpers.py:8: in <module>
    import ivy
/opt/project/ivy/__init__.py:770: in <module>
    from . import stateful
/opt/project/ivy/stateful/__init__.py:1: in <module>
    from . import activations
/opt/project/ivy/stateful/activations.py:5: in <module>
    from ivy.stateful.module import Module
/opt/project/ivy/stateful/module.py:8: in <module>
    import dill
E   ModuleNotFoundError: No module named 'dill'

Process finished with exit code 4

Empty suite`

2) I used ivy.histogram but while checking the implementation in jax backend, I found it in
 ivy/functional/backends/jax/experimental/statistical.py
 and didn't find it in
 ivy/functional/backends/jax/statistical.py, 
 so I am not sure is that normal or not?
 
 3) Same for _histogram_helper(), I used this function in while creating the unit test but I have imported it like this:
 `from ivy_tests.test_ivy.test_functional.test_experimental.test_core.test_statistical import (
    _histogram_helper,
)`
and coudn't find it in: `ivy_tests.test_ivy.test_functional.test_experimental.test_core.test_statistical`

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by marwanWaly** [comment]
Sorry I forget it.
Close #19652

File: ivy/functional/frontends/jax/numpy/statistical.py
Comment (theRealBird) [inline ivy/functional/frontends/jax/numpy/statistical.py:1]: We can remove the code which specifies the arg types. (Eg:- Union[ivy.Array, ivy.NativeArray] etc for `a`). You can refer to other implementations in this file for reference too. Thanks

**Comment by theRealBird** [comment]
Hi @marwanWaly,
Thank you so much for your contribution to Ivy and for your patience with the PR review process. To answer your questions:- 
1. I am not too sure about the error with "dill". I was unable to replicate it at my end. I am not sure if you tried this but maybe try installing the dill package using pip in your environment? That seems to work for me usually.

2. Yeah we shouldn't worry about the helpers being in experimental. It's just how we have structured the project. 

At the moment, the CI dashboard is telling me that, there are a lot of new test failures being introduced. They don't seem to be related to your code but I would suggest you to have a look at them once more.

Regarding the actual implementation, everything looks great except maybe one thing which I have requested to be changed. Thank you so much!

File: ivy/functional/frontends/jax/numpy/statistical.py
Comment (theRealBird) [reply ivy/functional/frontends/jax/numpy/statistical.py:1]: Please also remove the merge conflicts @marwanWaly. Thanks!

**Pull request closed without merging**